src/mcp/emulator/dataconnectEmulatorController.ts

```diff
@@ -0,0 +1,40 @@
+import { EmulatorHubClient } from "../../emulator/hubClient.js";
+import { EmulatorInfo, Emulators } from "../../emulator/types.js";
+
+export interface DataConnectEmulatorDetails {
+  host: string;
+  port: number;
+  url: string;
+}
+
+function formatEndpoint(emulatorInfo: EmulatorInfo): DataConnectEmulatorDetails {
+  const host = emulatorInfo.host.includes(":") ? `[${emulatorInfo.host}]` : emulatorInfo.host;
+  return {
+    host: emulatorInfo.host,
+    port: emulatorInfo.port,
+    url: `http://${host}:${emulatorInfo.port}`,
+  };
+}
+
+export async function getDataConnectEmulatorDetails(
+  hubClient?: EmulatorHubClient,
+): Promise<DataConnectEmulatorDetails | undefined> {
+  if (!hubClient || !hubClient.foundHub()) {
+    return undefined;
+  }
+
+  try {
+    const emulators = await hubClient.getEmulators();
+    const dcEmulatorInfo = emulators[Emulators.DATACONNECT];
+
+    if (dcEmulatorInfo) {
+      return formatEndpoint(dcEmulatorInfo);
+    }
+  } catch (error) {
+    console.warn(
```

Comment:
`console.warn` won't work well in MCP server.

cc @jonathanedey Do you know how to produce a good debugging logs in the MCP server?

Reply:
Use the `logger` - when running in `--debug` mode it should be capturing things in `firebase-debug.log`.